sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -126,6 +127,15 @@ class ParquetFileFormat
     // Sets compression scheme
     conf.set(ParquetOutputFormat.COMPRESSION, parquetOptions.compressionCodecClassName)
 
+    // Sets Zstd level and Zstd workers, only effect when compression codec is `zstd`
+    conf.set(
```

Comment:
You can use `setIfUnset` I believe. The main problem is that I am not sure if it's worthwhile. For example, we don't expose a configuration for Parquet writer v2 (e.g., `parquet.writer.version`).

Just for clarification, you are already able to set this via, e.g.:

```
    val spark = SparkSession.
      builder()
      .master("local[*]")
      .appName("AppName")
      .config("spark.sql.parquet.compression.codec", "zstd")
      .getOrCreate() 
    spark.sessionState.newHadoopConfWithOptions(   
      "parquet.compression.codec.zstd.level" -> "10",
      "parquet.compression.codec.zstd.workers" -> "0"
    )
    val csvfile = spark.read.csv("file:///home/test_data/Reviews.csv")
    csvfile.coalesce(1).write.parquet("file:///home/test_data/nn_parq_10")
```

I believe?

Reply:
+1 for @HyukjinKwon 's comment. The PR claim seems to be wrong because we can already.
> Unable to set zstd compression level while writing

Reply:
@HyukjinKwon @dongjoon-hyun 

In `ParquetOptions`, the `spark.sql.parquet.compression.codec` parameter also has a corresponding `parquet.compression` in parquet.

For zstd level & zstd workers, maybe we can expose these parameters for different priority configuration?

Reply:
We can do but what I am not sure if this is worthwhile. There are a lot of configurations that are not controlled by Spark conf e.g., Parquet writer version. Why are these special?

Reply:
OK, I see what you mean and agree with you.I will close this pr later.

Reply:
> You can use `setIfUnset` I believe. The main problem is that I am not sure if it's worthwhile. For example, we don't expose a configuration for Parquet writer v2 (e.g., `parquet.writer.version`).
> 
> Just for clarification, you are already able to set this via, e.g.:
> 
> ```
>     val spark = SparkSession.
>       builder()
>       .master("local[*]")
>       .appName("AppName")
>       .config("spark.sql.parquet.compression.codec", "zstd")
>       .getOrCreate() 
>     spark.sessionState.newHadoopConfWithOptions(   
>       "parquet.compression.codec.zstd.level" -> "10",
>       "parquet.compression.codec.zstd.workers" -> "0"
>     )
>     val csvfile = spark.read.csv("file:///home/test_data/Reviews.csv")
>     csvfile.coalesce(1).write.parquet("file:///home/test_data/nn_parq_10")
> ```
> 
> I believe?

Hi @HyukjinKwon, I believe the provided way works.
However, there is no way to modify the `parquet.compression.codec.zstd.{level|worker}` configs dynamically during spark runtime, right?

In my usage, I have a long-running spark app, that needs to increase the compression level if some conditions are met.

Do you guys have any suggestions?

Some example code in spark-shell
```
// bin/spark-shell
val f = spark.read.parquet("file:///tmp/input_imp_join_all.zstd.parquet")
f.sparkSession.sessionState.conf.setConfString("parquet.compression.codec.zstd.level", "3")
f.coalesce(1).write.parquet("file:///tmp/exp_level3")
f.sparkSession.sessionState.conf.setConfString("parquet.compression.codec.zstd.level", "6")
f.coalesce(1).write.parquet("file:///tmp/expp_level6") // this won't be level 6
```